ruby/ql/lib/codeql/ruby/Concepts.qll

```diff
@@ -304,6 +314,14 @@ module Http {
        * This is typically the name of the method that gives rise to this input.
        */
       string getSourceType() { result = super.getSourceType() }
+
+      /**
+       * Gets the kind of the accessed input,
+       * Can be one of "parameter", "header", "body", "url", "cookie".
+       *
+       * Note that this predicate is functional.
```

Comment:
What is the purpose of this `Note` ? I guess `functional` in this context indicates that the predicate returns exactly one result. This is kind of implied in the name of the predicate `get` instead of `getA` and the `Gets the` and lack of `if any` suffix in the comment. 

I think the note may confuse users and make them wonder if there are any predicates that do not work/function somehow.

Reply:
I'm going to back out of the `kind` related changes here as https://github.com/github/codeql/pull/10602 implements this in a neater way. FWIW this was more or less copied from the JS version of this, my understanding of `functional` in this sense was that the return values could be relied on to have a precise semantic meaning. This is in contrast to `string getSourceType()` which is a bit fuzzier in what it might return.

Reply:
> Sorry for conflicting with your changes here! When we talked about it in the meeting, I misunderstood and thought your changes were already merged, so when I couldn't find them I assumed they were for a different concept and so I would have to add them in my PR.

No worries, thanks for the improved implementation. I've updated this PR to use it.